01_financial-well-being/income_metrics_county.qmd

```diff
@@ -0,0 +1,735 @@
+---
+title: "Opportunity for Income Predictor, county-level"
+author: "JP Walsh"
+date: today
+format:
+  html:
+    toc: true
+    toc_float: true
+    embed-resources: true
+    code-fold: show
+execute: 
+  warning: false
+editor_options: 
+  chunk_output_type: console
+---
+
+*Program for the creation of the Opportunity for Income Predictor at the county level*
+
+ACS Code: Initially written by Tina Chelidze and updated by JP Walsh 2022-2023.
+
+Primary data derived from the IPUMS API.
+
+Based on processes developed by Paul Johnson and Kevin Werner in SAS.
+
+-   [Housekeeping](#housekeeping)
+-   [Read Data](#read-data)
+-   [Clean Data](#clean-data)
+-   [Merge Crosswalk](#merge-crosswalk)
+-   [Create Income Metric](#create-income-metric)
+-   [Evaluate Results](#evaluate-results)
+-   [Export](#export)
+
+## Housekeeping
+
+Read in packages for use. If you do not have an IPUMS API key you will need to make one via the [IPUMS website](https://www.ipums.org/).
+
+```{r}
+library(tidyverse)
+library(Hmisc)
+library(ipumsr)
+library(reactable)
+library(srvyr)
+
+options(scipen = 999)
+
+#set_ipums_api_key("Your KEY", save = TRUE)
```

Comment:
Although it isn't particularly problematic in this case, a more secure way of doing this would be to define the API key in an environment that is blocked from merging by .gitignore. Not imperative but I figured I would mentioned. API keys generally shouldn't be pushed to git, so it might be helpful to add a comment that says #DO NOT PUSH THIS AND YOU ONLY NEED TO RUN ONCE  https://cran.r-project.org/web/packages/httr/vignettes/secrets.html